packages/sql-mapper/lib/queries/mysql.js

```diff
@@ -24,26 +24,54 @@ function insertOne (db, sql, table, schema, input, primaryKey, useUUID, fieldsTo
     sql`, `
   )
 
-  return db.tx(async function (db) {
-    const insert = sql`
+  if (primaryKeys.length === 1 && input[primaryKeys[0].key] === undefined) {
+    return db.tx(async function (db) {
+      const insert = sql`
       INSERT INTO ${tableName(sql, table, schema)} (${keys})
       VALUES(${values})
     `
-    await db.query(insert)
+      await db.query(insert)
 
-    const res2 = await db.query(sql`
+      const res2 = await db.query(sql`
       SELECT ${sql.join(fieldsToRetrieve, sql`, `)}
       FROM ${tableName(sql, table, schema)}
-      WHERE ${sql.ident(primaryKey)} = (
+      WHERE ${sql.ident(primaryKeys[0].key)} = (
         SELECT last_insert_id()
       )
     `)
 
-    return res2[0]
-  })
+      return res2[0]
+    })
+  } else {
+    const where = []
+    for (const { key } of primaryKeys) {
+      // TODO write a test that cover this
+      /* istanbul ignore next */
+      if (!input[key]) {
+        throw new Error(`Missing value for primary key ${key}`)
+      }
+      where.push(sql`${sql.ident(key)} = ${input[key]}`)
+    }
+
+    return db.tx(async function (db) {
+      const insert = sql`
+      INSERT INTO ${sql.ident(table)} (${keys})
+      VALUES(${values})
+    `
+      await db.query(insert)
+
+      const res2 = await db.query(sql`
+        SELECT ${sql.join(fieldsToRetrieve, sql`, `)}
+        FROM ${sql.ident(table)}
```

Comment:
Same here. we might add a test for these cases when using schemas to catch these issues